### PR TITLE
Filtered upsampling

### PR DESCRIPTION
**Pull request by yikwill**
# PhysicsNeMo Pull Request
This PR adds a new upsampling scheme. Transposed convolutions imprint a well-known checkerboarding pattern. Additionally, simple nearest neighbor and bilinear upsampling methods applied to HEALPix faces introduce artifacts at seams and along lines of constant latitude when the incoming signal is zonally uniform (no east-west variance). This new upsampling scheme mitigates this by applying padding before the interpolation and also applying a HEALPix specific four-point smoother after the interpolation which preserves zonal uniformity when the interpolation mode is nearest neighbor or bilinear.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 1]
```diff
@@ -1245,3 +1348,68 @@ def forward(self, x: th.Tensor) -> th.Tensor:
...
+        self.scale_factor = scale_factor
+        self.mode = mode
+        self.trim_size = trim_size
+        self.interp = th.nn.functional.interpolate
+
+        # Four point smoother specific to HPX grid. This smooths out the specific
+        # type of aliasing that nearest neighbor and bilinear upsampling introduce
+        # into zonally uniform signals
+        self.smoother_kernel = torch.tensor(
+            [[0.,1.,0.],
```
Comment (daviddpruitt) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:1392]: Should these be set to 0.25 so the total adds up to 1 and you avoid the division during the forward step?

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 2]
```diff
@@ -1161,6 +1161,109 @@ def forward(self, x):
...
+            Activation function used in upsampling
+        enable_nhwc: bool, optional
+            Enable nhwc format, passed to wrapper
+        enable_healpixpad: bool, optional
+            If HEALPixPadding should be enabled, passed to wrapper
+        """
+        super().__init__()
+
+        if dilation > 1:
+            raise Exception(
```
Comment (daviddpruitt) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: This should be narrowed down to ValueError rather than the more generic Exception

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 1 above)
Comment (yikwill) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:1392]: Good idea. I made a change so that the kernel is normalized to sum to 1 during the init.

**Review by daviddpruitt — APPROVED** [review]
